mlpp_lib/normalizers.py

```diff
@@ -0,0 +1,519 @@
+            )
+
+        self.mean = dataset[variables].mean(dims).compute().copy()
+        self.std = dataset[variables].std(dims).compute().copy()
+        self.fillvalue = self.fillvalue
+        # Check for near-zero standard deviations and set them equal to one
+        self.std = xr.where(self.std < 1e-6, 1, self.std)
+
+    def transform(
+        self, *datasets: xr.Dataset, variables: Optional[list] = None
+    ) -> tuple[xr.Dataset, ...]:
+        if self.mean is None:
+            raise ValueError("Standardizer wasn't fit to data")
+
+        def f(ds: xr.Dataset, variables: Optional[list] = None):
+
+            ds_copy = ds.copy()
+            if variables is None:
+                variables = list(ds_copy.data_vars)
+            for var in variables:
+                assert var in self.mean.data_vars, f"{var} not in Standardizer"
+
+                standardized_var = (
+                    (ds_copy[var] - self.mean[var]) / self.std[var]
+                ).astype("float32")
+                if self.fillvalue is not None:
+                    standardized_var = standardized_var.fillna(self.fillvalue)
+
+                ds_copy[var] = standardized_var
+
+            return ds_copy
+
+        return tuple(f(ds, variables) for ds in datasets)
+
+    def inverse_transform(
+        self, *datasets: xr.Dataset, variables: Optional[list] = None
+    ) -> tuple[xr.Dataset, ...]:
+        if self.mean is None:
+            raise ValueError("Standardizer wasn't fit to data")
+
+        def f(ds: xr.Dataset, variables: Optional[list] = None) -> xr.Dataset:
+            if variables is None:
+                variables = ds.data_vars
+
+            ds = xr.where(ds > self.fillvalue, ds, np.nan)
+            for var in variables:
+                assert var in self.mean.data_vars, f"{var} not in Standardizer"
+
+                unstandardized_var = (ds[var] * self.std[var] + self.mean[var]).astype(
+                    "float32"
+                )
+                ds[var] = unstandardized_var
+
+            return ds
+
+        return tuple(f(ds, variables) for ds in datasets)
+
+    @classmethod
+    def from_dict(cls, in_dict: dict) -> Self:
+        mean = xr.Dataset.from_dict(in_dict["mean"])
+        std = xr.Dataset.from_dict(in_dict["std"])
+        fillvalue = in_dict["fillvalue"]
+        return cls(mean, std, fillvalue=fillvalue)
+
+    def to_dict(self):
+        out_dict = {
+            "mean": self.mean.to_dict(),
+            "std": self.std.to_dict(),
+            "fillvalue": self.fillvalue,
+        }
+        return out_dict
+
+
+@dataclass
+class MinMaxScaler(DataTransformation):
+    """
+    Tranforms data using a min/max scaling in a xarray.Dataset object.
+    """
+
+    minimum: xr.Dataset = field(default=None)
+    maximum: xr.Dataset = field(default=None)
+    fillvalue: dict[str, float] = field(init=True, default=-5)
```

Comment:
we could set the default fillvalue to -1 here, what do you think?

Reply:
Yes it makes sense and should prevent too abrupt changes in the loss